pytest_dependency.py

```diff
@@ -59,10 +61,15 @@ def addResult(self, item, marker, rep):
         status = self.results.setdefault(name, DependencyItemStatus())
         status.addResult(rep)
 
-    def checkDepend(self, depends):
+    def checkDepend(self, depends, item):
         for i in depends:
             if not(i in self.results and self.results[i].isSuccess()):
-                pytest.skip("depends on %s" % i)
+                msg = "%s depends on %s (failed)" % (item.name, i)
```

Comment:
That is claiming too much.  If `isSuccess()` yields `False` that could either mean the dependency has not been run at all, it has been skipped, or it has failed.  You would need to check that before claiming it was a fail.

Reply:
The initial approach was to show the test names involved when something wasn't right, so I simplified the skip's reason message and remove the commented code.